### PR TITLE
Add post and comment models

### PR DESCRIPTION
**Pull request by imkingdavid**

### CODE REVIEW

File: blog/model/comment.php [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	 * @return bool
+	 */
+	public function disapprove($id)
+	{
+		$sql = 'UPDATE ' . $this->blog_comments_table . ' SET status = 0 WHERE id = ' . (int) $id;
+		$this->db->sql_query($sql);
+
+		return true;
+	}
+}
```
Comment (michaelcullum) [inline blog/model/comment.php:176]: We need a load data for this, one that gets an individual record, one that gets an optional number of records, with optional asc and desc and optional WHERE and one that gets all records

File: blog/model/post.php [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+			'locked' => (bool) $locked,
+			'poster_id' => (int) $poster_id,
+			'content' => $content,
+			'bbcode_uid' => $bbcode_uid,
+			'bbcode_bitfield' => $bbcode_bitfield,
+		));
+		$this->db->sql_query($sql);
+		$post_id = (int) $this->db->sql_next_id();
+
+		foreach ($categories as $category_id)
```
Comment (michaelcullum) [inline blog/model/post.php:107]: This belongs separately.

File: blog/model/post.php [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+				'post_id' => (int) $post_id,
+				'category_id' => (int) $category_id,
+			));
+			$this->db->sql_query($sql);
+
+			$sql = 'UPDATE ' . $this->blog_categories_table . ' SET post_count = post_count + 1 WHERE id = ' . (int) $category_id;
+			$this->db->sql_query($sql);
+		}
+
+		foreach ($tags as $tag_id)
```
Comment (michaelcullum) [inline blog/model/post.php:119]: should also be separate

**Comment by michaelcullum** [comment]
My comments sort of apply to the whole thing. A method should only do one thing. So it might update a post, it might create a post, it might get a post's data, it might get all post data, but it shouldn't do the other things like creating a record in the table which links tags and posts or categories and posts. That is logic and doesn't belong here. Also the result should be returned by the methods.

**Comment by michaelcullum** [comment]
The idea of a repository is so if db schema changes, we don't have to change it throughout the application so much as just in this one place, as soon as logic gets involved it becomes an operation.

**Comment by imkingdavid** [comment]
So in other words, a method inside one of these models should only do exactly what it says. It shouldn't modify anything related to it. For instance, when deleting a post, at some point we need to also delete comments, decrease the post count on the categories to which the post is assigned, and remove the relations between categories/tags and the post. But in the model, all that should be done is deleting the record from the posts table, and we will add the other operations in other classes.

I guess that makes sense, but at some point we should make an operation class that actually brings all of those individual operations together so that we can still do $post->delete(); and only have to call the one line of code instead of several each time we delete a post.

**Comment by michaelcullum** [comment]
Yup, we will. The controller calls things like $post->delete($id) which calls the operation which will then call all the relevant stuff in the model. Separation of concerns

**Comment by michaelcullum** [comment]
@imkingdavid See michaelcullum@bd8f974 if you want to see what i mean by an operator. It's just an interface there to remind me of the methods i'm calling in controllers and the arguments but it gives you an idea. They touch many of these different respositories (i.e. creating a post would create the post, create tag links etc.)

**Pull request merged**